### PR TITLE
Assign 'lang="uk"' to STY secondary column components with a language override

### LINKED ISSUE #7264: Assign 'lang="uk"' to relevant Ukrainian STY page components when a language override is present
### Is your feature request related to a problem? Please describe.
In order to support Ukrainian in Russian, a Russian language code will be applied to the root element (`<html>`) in https://github.com/bbc/simorgh/issues/7262. However, some areas of the page will remain in Ukrainian and must have their language codes fixed to Ukrainian.

### Describe the solution you'd like
When a language override is present on BBC Ukrainian STYs, ensure the Header (nav), Secondary Column, and Footer, all have their language code fixed to [headerFooterLang](https://github.com/bbc/simorgh/blob/6383e745986eba7575c51d1ca88f64e3e6aa4764/src/app/lib/config/services/ukrainian.js#L288) in the service context. We might choose to update the name of this property also.

Ensure this is covered by integration tests.

#### A11y acceptance criteria
```
Given I’m using a screen reader
When I navigate the header (nav) column with Ukrainian language content
Then it is announced with a Ukrainian accent

Given I’m using a screen reader
When I navigate the secondary column with Ukrainian language content
Then it is announced with a Ukrainian accent

Given I’m using a screen reader
When I navigate the footer with Ukrainian language content
Then it is announced with a Ukrainian accent
```

Note: @OlgaLyubin will be able to help us with the dev a11y swarm.

### Describe alternatives you've considered
n/a

### Testing notes
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

### Additional context
- View the source of https://www.bbc.com/ukrainian/ukraine_in_russian to see an example of a page where we already do this.

### PR DESCRIPTION
**Pull request by Simon2828**
Resolves #7264

**Overall change:**
Fix the language code to Ukrainian for the secondary column components on StoryPage.

**Code changes:**

- Add secondary column lang attribute to StoryPage
- Rename `headerFooterLang` property

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ X I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [X] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/StoryPage/index.test.jsx [hunk 1]
```diff
@@ -24,6 +24,9 @@ import pidginSecondaryColumnData from '#data/pidgin/secondaryColumn/index.json';
 import igboPageData from '#data/igbo/cpsAssets/afirika-23252735';
 import igboMostReadData from '#data/igbo/mostRead/index.json';
 import igboSecondaryColumnData from '#data/igbo/secondaryColumn/index.json';
+import ukrainianPageData from '#data/ukrainian/cpsAssets/news-russian-23333960.json';
```
Comment (rhenshaw56) [inline src/app/pages/StoryPage/index.test.jsx]: would it be worth renaming this as it's not just specific to `Ukranian` ?

**Review by rhenshaw56 — APPROVED** [review]

File: src/app/containers/Footer/index.jsx [hunk 2]
```diff
@@ -12,15 +12,15 @@ const getCopyrightText = text => (
...
-  const { footer, service, headerFooterLang } = useContext(ServiceContext);
+  const { footer, service, officialLang } = useContext(ServiceContext);
 
   if (!footer) {
     return null;
   }
 
-  // 'headerFooterLang' value is only available in the ukrainian config as our ukraine_in_russian pages will have
+  // 'officialLang' value is only available in the ukrainian config as our ukraine_in_russian pages will have
   // a ukrainian text for header and footer but a russian text main element
```
Comment (chris-hinds) [inline src/app/containers/Footer/index.jsx]:
```suggestion
  // 'officialLang' value is only available in the ukrainian config as our ukraine_in_russian pages will have
  // ukrainian text for the header and footer but russian text for the main element
```

File: src/app/containers/Header/index.jsx [hunk 3]
```diff
@@ -31,10 +31,10 @@ const HeaderContainer = () => {
 
   const isOperaMini = onClient() && window.operamini;
 
-  // 'headerFooterLang' value is only available in the Ukrainian config as our ukraine_in_russian pages will have
+  // 'officialLang' value is only available in the Ukrainian config as our ukraine_in_russian pages will have
   // a Ukrainian text for the header and footer but a Russian text for the main element.
   // However, the skip to content link will also be in Russian, so we need to pass the `ru-UA` lang code to it.
```
Comment (chris-hinds) [inline src/app/containers/Header/index.jsx]:
```suggestion
  // 'officialLang' value is only available in the Ukrainian config as our ukraine_in_russian pages will have
  // Ukrainian text for the header and footer but Russian text for the main element.
  // However, the skip to content link will also be in Russian, so we need to pass the `ru-UA` lang code to it.
```

File: src/app/pages/StoryPage/index.jsx [hunk 4]
```diff
@@ -84,6 +85,9 @@ const StoryPage = ({ pageData, mostReadEndpointOverride }) => {
   );
   const featuresInitialData = path(['secondaryColumn', 'features'], pageData);
   const recommendationsInitialData = path(['recommendations'], pageData);
+  // 'officialLang' value is only available in the Ukrainian config as the ukraine_in_russian pages have
+  // a Ukrainian text for the secondary column but a Russian text for the main element.
```
Comment (chris-hinds) [inline src/app/pages/StoryPage/index.jsx]:
```suggestion
  // 'officialLang' value is only available in the Ukrainian config as the ukraine_in_russian pages have
  // Ukrainian text for the secondary column but Russian text for the main element.
```

**Review by chris-hinds — CHANGES_REQUESTED** [review]

File: src/app/pages/StoryPage/index.test.jsx [hunk 5]
```diff
@@ -38,17 +41,18 @@ jest.mock('#containers/ChartbeatAnalytics', () => {
...
-      <ServiceContextProvider service={service}>
+      <ServiceContextProvider service={service} variant={variant}>
         <RequestContextProvider
           bbcOrigin="https://www.test.bbc.co.uk"
           isAmp={false}
           pageType={pageData.metadata.type}
           pathname={pageData.metadata.locators.assetUri}
           service={service}
           statusCode={200}
+          variant={pageData.variant}
```
Comment (chris-hinds) [inline src/app/pages/StoryPage/index.test.jsx]: should this be `pageData.variant` when a `variant` prop is passed in?

Comment (Simon2828) [reply src/app/pages/StoryPage/index.test.jsx]: yes this is not needed 👍

**Review by chris-hinds — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
Testing:
URL --> http://localhost:7080/ukrainian/news-russian-23333960 

The header footer and secondary column are fixed to Ukranian ?

LGTM..

Accessibility is still pending..

**Comment by Simon2828** [comment]
@OlgaLyubin checked the a11y acceptance criteria with VoiceOver but as the Ukrainian language is not supported, it only read Russian correctly. For Ukrainian it was reading: “Cyryllic capital letter O, cyryllic small letter m…” and so on. So for the full a11y swarm we will need to record videos for her to check, using screen readers that do support the Ukrainian language.

**Review by chris-hinds — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

**Review by simonsinclair — APPROVED** [review]

**Comment by Simon2828** [comment]
> @OlgaLyubin checked the a11y acceptance criteria with VoiceOver but as the Ukrainian language is not supported, it only read Russian correctly. For Ukrainian it was reading: “Cyryllic capital letter O, cyryllic small letter m…” and so on. So for the full a11y swarm we will need to record videos for her to check, using screen readers that do support the Ukrainian language.

This will be included in the full a11y swarm: https://github.com/bbc/simorgh/issues/7442

**Pull request merged**